### DISCUSSION TITLE
Best way to read multiple node values at once together with the source timestamp

### DISCUSSION THREAD

**Discussion by nkawerau** [FreeOpcUa/opcua-asyncio#1337] [Q&A]
I am aware that there is the great function read_values() but it returns only the value of each node.
It also exsists the function  read_data_value() it returns the DataValue (Value + StatusCode + Timestamps) for one node.

It does not exsist a function read_data_values() which returns the value and timestamp for multiple nodes at once?

Or is there another best practice solution to get value and timestamp for multiple nodes at once?

**Comment by AndreasHeine** [comment]
https://github.com/AndreasHeine/opcua-tutorial/blob/main/client/read.py

https://github.com/AndreasHeine/opcua-tutorial/blob/8f1dc98e8f2301514aff5fd83ea31fdb35c82029/client/read.py#L34
